src/Model/Request/ReferenceFile.php

```diff
@@ -30,20 +29,17 @@ public function setFile(File $file): ReferenceFile
         return $this;
     }
 
-    /**
-     * @return string[]
-     */
-    public function getReferenceLanguages(): array
+    public function getReferenceLanguages(): StringCollection
     {
         return $this->referenceLanguages;
     }
 
     /**
-     * @param string[] $referenceLanguages
+     * @param \OpenEuropa\CdtClient\Model\StringCollection|array<int, string> $referenceLanguages
      */
-    public function setReferenceLanguages(array $referenceLanguages): ReferenceFile
+    public function setReferenceLanguages(StringCollection|array $referenceLanguages): self
```

Comment:
normally if there are more use cases you have more methods, ex 
setReferenceLanguages and setReferenceLanguagesFromArray,

what are the 2 uses cases, when do we expect one and not the other?

In case we don't have distinct situations and we cant plan to call a separate method, then we could move this logic into the BaseCollection, so that the constructor excepts also a instance of the child implementation
something in the lines of 
```
public function __construct(array|self $data....
...
if ($data instanceof static) {  // return, its already an object of this type
  return $this;
}
// go on validating etc
```

so after we either have 2 typecast methods or one flexible one, with a mixed parameter (to not go creazy) and we know the collection takes care of the validation

Reply:
I will try to optimize it, but as far as I can remember, the approach was forced by both the serializer and PHPStan.

Reply:
Actually, you're right, I kept it from the previous version, and the array version can be simplified